master/internal/rm/kubernetesrm/spec.go

```diff
@@ -361,6 +369,19 @@ func (p *pod) configurePodSpec(
 	if podSpec.ObjectMeta.Labels == nil {
 		podSpec.ObjectMeta.Labels = make(map[string]string)
 	}
+	// add determined metadata as pod labels
+	podSpec.ObjectMeta.Labels[userLabel] = p.submissionInfo.taskSpec.Owner.Username
```

Comment:
In other places in the code we treat Owner as nullable. 

https://github.com/determined-ai/determined/blob/5a588e0f00b2beef87d6d73519b1de67e279405d/master/pkg/tasks/task.go#L202

I'm 95% sure that Owner can't be null in the current code version but I don't think it is worth the risk of a customer issue so I would say add a check that Owner is not null. 

We can create a follow up ticket if you want to refactor the code in a way that is more explicit that Owner won't be nil.